modules/indexing/src/main/java/org/apache/ignite/internal/visor/dr/VisorDrCheckPartitionCountersTask.java

```diff
@@ -0,0 +1,291 @@
+    private static class DrCacheEntryValidateJob
+            extends VisorJob<VisorDrCheckPartitionCountersTaskArg, Collection<VisorDrCheckPartitionCountersJobResult>> {
+        /** Serial number */
+        private static final long serialVersionUID = 0L;
+
+        /** */
+        private final Map<String, Set<Integer>> cachesWithPartitions;
+
+        /**
+         * Create job with specified argument.
+         *
+         * @param arg Last time when metrics were collected.
+         * @param debug Debug flag.
+         */
+        protected DrCacheEntryValidateJob(@Nullable VisorDrCheckPartitionCountersTaskArg arg, Map<String, Set<Integer>> cachesWithPartitions, boolean debug) {
+            super(arg, debug);
+
+            this.cachesWithPartitions = cachesWithPartitions;
+        }
+
+        /** {@inheritDoc} */
+        @Override protected Collection<VisorDrCheckPartitionCountersJobResult> run(
+                @Nullable VisorDrCheckPartitionCountersTaskArg arg
+        ) throws IgniteException {
+            assert arg != null;
+
+            int checkFirst = arg.getCheckFirst();
+
+            List<VisorDrCheckPartitionCountersJobResult> metrics = new ArrayList<>();
+
+            for (Entry<String, Set<Integer>> cachesWithParts : cachesWithPartitions.entrySet()) {
+                metrics.add(
+                        calculateForCache(cachesWithParts.getKey(), cachesWithParts.getValue(), checkFirst)
+                );
+            }
+
+            return metrics;
+        }
+
+        private VisorDrCheckPartitionCountersJobResult calculateForCache(String cache, Set<Integer> parts, int checkFirst) {
+            ignite.cache(cache);
+
+            CacheGroupContext grpCtx = ignite.context().cache().cacheGroup(CU.cacheId(cache));
+
+            if (grpCtx == null)
+                grpCtx = ignite.cachex(cache).context().group();
+
+            int size = 0;
+            int entriesProcessed = 0;
+            int brokenEntriesFound = 0;
+            Set<Integer> affectedCaches = new HashSet<>();
+            Set<Integer> affectedPartitions = new HashSet<>();
+
+            for (Integer part : parts) {
+                DrCachePartitionMetrics partMetrics = calculateForPartition(grpCtx, cache, part, checkFirst);
+
+                size += partMetrics.getSize();
+                entriesProcessed += partMetrics.getEntriesProcessed();
+
+                if (partMetrics.getBrokenEntriesFound() > 0) {
+                    affectedCaches.addAll(partMetrics.getAffectedCaches());
+                    affectedPartitions.add(part);
+                    brokenEntriesFound += partMetrics.getBrokenEntriesFound();
+                }
+            }
+
+            return new VisorDrCheckPartitionCountersJobResult(cache, size, affectedCaches, affectedPartitions, entriesProcessed, brokenEntriesFound);
+        }
+
+        private DrCachePartitionMetrics calculateForPartition(CacheGroupContext grpCtx, String cache, int part, int checkFirst) {
```

Comment:
I cannot catch an idea with check N first entries. What is this feature for? Reduce check workload?
What the user should do when N first entries don't contain errors?
Maybe the flags `findUntilFirstError` or integer parameter `findUntilFirstsError` make sense?